### ISSUE TITLE
`no-unused-vars` rule does not recognize function parameter used with spread syntax

### ISSUE THREAD

**Issue by puresick** [eslint/eslint#11549]
**Tell us about your environment**
Arch Linux
Linux 5.0.4

* **ESLint Version: 5.9.0**
* **Node Version: v10.15.1**
* **npm Version: 6.4.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function (parameter) {
  const { value } = parameter

  const anotherObject = {
    ...value,
    anotherKey: anotherValue
  }
}
```

```bash
npx eslint app/** --ext .js,.vue
```

**What did you expect to happen?**
That eslint will recognize the function parameter `parameter` is being used.

**What actually happened? Please include the actual, raw output from ESLint.**
The `no-unused-vars` rule triggers an error that `parameter` is not being used despite the fact that it is being used while destructuring with the spread syntax.

**Are you willing to submit a pull request to fix this bug?**
If this will be identified as a bug I could try fixing the problem and submit a pull request.

EDIT: corrected an error in my code example above

**Comment by not-an-aardvark** [comment]
I think you want to use the [`ignoreRestSiblings`](https://eslint.org/docs/rules/no-unused-vars) option.

**Comment by nzakas** [comment]
@puresick can you include the actual output you're seeing?

As it is, I'm unable to reproduce. I do see a warning that says `value` isn't used, and that is correct, but no warning for `parameter`.

**Comment by puresick** [comment]
> @puresick can you include the actual output you're seeing?
> 
> As it is, I'm unable to reproduce. I do see a warning that says `value` isn't used, and that is correct, but no warning for `parameter`.

Oh there was a fault on my site, I am sorry!
Of course I want to use `value` with the spread syntax, not `parameter` - I will correct it in the first post.

The correct example would than look like this: 

```
function (parameter) {
  const { value } = parameter

  const anotherObject = {
    ...value,
    anotherKey: anotherValue
  }
}
```

Like @not-an-aardvark told one post before, I should use the `ignoreRestSiblings` option to prevent the error. But isn't it still wrong if it is recognized as unused despite that it is being used with the spread syntax? If this should be the way, can you provide me some details on why it is behaving like this?

Thank you! :)

**Comment by platinumazure** [comment]
Closing as this question has been answered. Please feel free to stop by our [Gitter chat](https://gitter.im/eslint/eslint) with any further questions. Thanks!
